euphoria-flink/src/main/java/cz/seznam/euphoria/flink/batch/BatchStateStorageProvider.java

```diff
@@ -140,22 +141,34 @@ public void add(T element) {
 
     @Override
     public Iterable<T> get() {
+      return this::iter;
+    }
 
+    private Iterator<T> iter() {
+      FileInputStream finput;
```

Comment:
Maybe it is not necessary to store `finput` explicitly just to call close on it. Now the `FileInputStream` is actually closed twice. Once by `Closeables.closeQuietly(finput)` and second transitively from `input.close()`.
But it does not cause any harm.